core/src/main/java/org/elasticsearch/index/query/BoolQueryParser.java

```diff
@@ -78,12 +78,12 @@ public BoolQueryBuilder fromXContent(QueryParseContext parseContext) throws IOEx
                     shouldClauses.add(query);
                     break;
                 case "filter":
-                    query = parseContext.parseInnerFilterToQueryBuilder();
+                    query = parseContext.parseInnerQueryBuilder();
```

Comment:
can you remind me what the replacement is for this isFilter flag? This doesn't need any adjustment because in the toQuery of these queries we already call toFilter ?